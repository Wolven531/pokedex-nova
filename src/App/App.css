:root {
	--spacing-gutter: 5px;
	--spacing-page-pad: 50px;
}

footer {
	align-items: center;
	background-color: rgba(25, 25, 25, 0.5);
	border-radius: 10px;
	display: flex;
	grid-column: 2;
	grid-row: 4;
	justify-content: center;
	text-align: center;
}

header {
	align-items: center;
	display: flex;
	grid-column: 2;
	grid-row: 1;
	justify-content: center;
	text-align: center;
}

header h2 {
	margin-block-start: 0;
	margin-block-end: 0;
	margin-inline-start: 0;
	margin-inline-end: 0;
}

section {
	grid-column: 2;
	grid-row: 3;
	text-align: center;
}

.app {
	background-color: rgba(50, 50, 50, 0.5);
	border-radius: 10px;
	column-gap: var(--spacing-gutter);
	display: grid;
	grid-template-columns:
		var(--spacing-page-pad)
		1fr
		var(--spacing-page-pad);
	grid-template-rows:
		var(--spacing-page-pad)
		var(--spacing-page-pad)
		1fr
		var(--spacing-page-pad)
		var(--spacing-page-pad);
	margin: 0;
	min-height: 400px;
	padding: 0;
	row-gap: var(--spacing-gutter);
}

.poke-container > a {
	text-transform: capitalize;
}
