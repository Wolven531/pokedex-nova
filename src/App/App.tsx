import React, { FC, useCallback, useEffect, useState } from 'react'
import { BrowserRouter, Route, Routes } from 'react-router-dom'
import { MainClient } from 'pokenode-ts'
// import { MainClient, NamedAPIResource } from 'pokenode-ts'
import { MAX_NUM_POKEMON, NUM_FORMATTER } from '../constants'
import { Layout } from '../Layout'
import './App.css'

export const App: FC = () => {
	const api = new MainClient()

	const [isLoaded, setIsCountLoaded] = useState(false)
	// const [pokemon, setPokemon] = useState<NamedAPIResource[]>([])
	const [totalCount, setTotalCount] = useState(0)

	/**
	 * This method uses the PokeAPI client to
	 *   - retrieve all pokemon entries
	 *   - update `totalCount` for rendering
	 *   - sets the component to loaded when successful
	 *
	 * If already loaded, this method simply resolves
	 */
	const fetchPokemonCount = useCallback(
		(): Promise<void> =>
			isLoaded
				? Promise.resolve()
				: api.pokemon
						.listPokemons(undefined, MAX_NUM_POKEMON)
						.then((result) => {
							setTotalCount(result.count)
							// setPokemon(result.results as NamedAPIResource[])
						})
						.catch((err) => {
							console.error(
								`Problem fetching pokemon count - ${err}`,
							)
						})
						.finally(() => {
							setIsCountLoaded(true)
						}),
		[isLoaded],
	)

	useEffect(() => {
		fetchPokemonCount()
	}, [fetchPokemonCount])

	return (
		<article className="app">
			<BrowserRouter>
				<Routes>
					<Route path="/" element={<Layout />}>
						<Route
							element={
								<>
									{!isLoaded && <div>Loading...</div>}
									{isLoaded && (
										<div>
											Total pokemon count -{' '}
											{NUM_FORMATTER.format(totalCount)}
										</div>
										// {pokemon.map(({ name, url }) => (
										// 	<div key={url} className="poke-container">
										// 		<a href={url}>{name}</a>
										// 	</div>
										// ))}
									)}
								</>
							}
							index
						></Route>
						<Route element={<div>Other</div>} path="other"></Route>
						{/* Not Found below */}
						<Route element={<div>Not Found</div>} path="*"></Route>
					</Route>
				</Routes>
			</BrowserRouter>
		</article>
	)
}
