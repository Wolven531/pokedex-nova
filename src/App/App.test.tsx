import React from 'react'
import { render, RenderResult, screen } from '@testing-library/react'
import { App } from './App'

describe('App', () => {
	let comp: RenderResult

	beforeEach(() => {
		comp = render(<App />)
	})

	it('renders header', () => {
		expect(comp.getByText(/Pokedex Nova/)).toBeInTheDocument()
	})

	it('matches snapshot', () => {
		expect(comp.asFragment()).toMatchSnapshot()
	})
})
