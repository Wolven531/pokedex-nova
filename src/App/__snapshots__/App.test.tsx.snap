// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<DocumentFragment>
  <article
    class="app"
  >
    <header>
      <h2>
        Pokedex Nova
      </h2>
    </header>
    <footer>
      Â© Anthony Williams 2022
    </footer>
  </article>
</DocumentFragment>
`;
